pkg/build/registry/build/strategy.go

```diff
@@ -67,7 +67,14 @@ func (strategy) PrepareForCreate(obj runtime.Object) {
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (strategy) PrepareForUpdate(obj, old runtime.Object) {
-	_ = obj.(*api.Build)
+	newBuild := obj.(*api.Build)
+	oldBuild := old.(*api.Build)
+	// Status can only be changed through the status subresource,
+	// EXCEPT if cancelling the build.
+	// TODO: Make the cancel request part of the spec
+	cancelled := newBuild.Status.Cancelled
+	newBuild.Status = oldBuild.Status
```

Comment:
No it won't ... the update meant to cancel a build will still be done the same way. And the controller doesn't do this update ... the client does.